source/meta.tex

```diff
@@ -2554,6 +2570,3461 @@
+
+\pnum
+\returns
+If \tcode{r} represents
+\begin{itemize}
+\item a non-static data member of type $T$,
+\item a data member description $(T, N, A, W, \mathit{NUA})$, or
+\item \tcode{dealias(r)} represents a type $T$,
+\end{itemize}
+then \tcode{sizeof($T$)} if $T$ is not a reference type
+and \tcode{size_of(\brk{}add_pointer(\brk{}\reflexpr{$T$}))} otherwise.
+Otherwise, \tcode{size_of(type_of(r))}.
+\begin{note}
+It is possible that while \tcode{sizeof(char)\brk{} == size_of(\reflexpr{char})}
+%FIXME: "is true"
+%FIXME: comma
+that \tcode{sizeof(char\&)\brk{} != size_of(\brk{}\reflexpr{char}\&)}.
+%FIXME: "is true"
+If \tcode{b} represents a direct base class relationship of an empty base class,
+then \tcode{size_of(b) > 0}.
+%FIXME: "is true"
+\end{note}
+\end{itemdescr}
+
+\indexlibraryglobal{alignment_of}%
+\begin{itemdecl}
+consteval size_t alignment_of(info r);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constantwhen
+\tcode{dealias(r)} is a reflection of a
+type,
+object,
+value,
+variable of non-reference type,
+non-static data member that is not a bit-field,
+direct base class relationship, or
+data member description.
+If \tcode{dealias(r)} represents a type,
+then \tcode{is_complete_type(r)} is \tcode{true}.
+
+\pnum
+\returns
+\begin{itemize}
+\item
+  If \tcode{dealias(r)} represents a type $T$,
+  then the alignment requirement for the layout-associated type\iref{class.mem.general}
+  for a non-static data member of type $T$.
+\item
+  Otherwise, if \tcode{dealias(r)} represents a variable or object,
+  then the alignment requirement of the variable or object.
+\item
+  Otherwise, if \tcode{r} represents a direct base class relationship,
+  then \tcode{alignment_of(type_of(r))}.
+\item
+  Otherwise, if \tcode{r} represents a non-static data member,
+  then the alignment requirement of the subobject
+  associated with the represented entity
+  within any object of type \tcode{parent_of(r)}.
+\item
+  Otherwise, \tcode{r} represents a data member description
+  $(\mathit{TR}, N, A, W, \mathit{NUA})$\iref{class.mem.general}.
+  If $A$ is not $\perp$,
+  then the value of $A$.
+  Otherwise, \tcode{alignment_of(\reflexpr{$T$})}.
+\end{itemize}
+\end{itemdescr}
+
+\indexlibraryglobal{bit_size_of}%
+\begin{itemdecl}
+consteval size_t bit_size_of(info r);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constantwhen
+\tcode{dealias(r)} is a reflection of a
+type,
+object,
+value,
+variable of non-reference type,
+non-static data member,
+unnamed bit-field,
+direct base class relationship, or
+data member description.
+If \tcode{dealias(r)} represents a type $T$,
+there is a point within the evaluation context from which $T$ is not incomplete.
+
+\pnum
+\returns
+\begin{itemize}
+\item
+  If \tcode{r} represents a non-static data member
+  that is a bit-field
+  %FIXME: comma. I believe non-static data members cannot be unnamed bit-fields,
+  %so the comma would prevent the sentence from suggesting that false reality.
```

Comment:
I'd prefer to just reverse the two prongs of the condition.
I'm also unhappy about the "with width $W$ not clearly attaching to both prongs.

Reply:
I like the idea of changing the order in the condition. I've fixed it up that way.